src/coreclr/gc/gc.cpp

```diff
@@ -17673,6 +17539,14 @@ int gc_heap::joined_generation_to_condemn (BOOL should_evaluate_elevation,
 #endif //BACKGROUND_GC
 #endif //STRESS_HEAP
 
+#ifdef BACKGROUND_GC
```

Comment:
Ah, so you just moved this code here from below?

Reply:
it fixes a problem as I mentioned in the checkin notes: 

> Moved the decision of condemning 1 when bgc is in progress to joined_generation_to_condemn (should logically belong there anyway) which fixes the problem that GCToEEInterface::DiagGCStart was getting the wrong condemned gen in this case.

Reply:
also it's just nicer to have all the code that makes the decision of which gen to condemn in only `generation_to_condemn` and `joined_generation_to_condemn`